sklearn/cluster/k_means_.py

```diff
@@ -186,7 +213,8 @@ def _check_sample_weight(X, sample_weight):
 def k_means(X, n_clusters, sample_weight=None, init='k-means++',
```

Comment:
How would these changes interact with https://github.com/scikit-learn/scikit-learn/pull/11950/, since both are modifying `k_means()`?

Reply:
I'm going for minimal changes to the current implementation. However, I looked into the cited pull request and I believe the cosine distance using chunked BLAS calls would work as well.